server/handler.go

```diff
@@ -335,6 +335,21 @@ func (h *Handler) ComBinlogDumpGTID(c *mysql.Conn, logFile string, logPos uint64
 
 var queryLoggingRegex = regexp.MustCompile(`[\r\n\t ]+`)
 
+func updateStatusVariableQuestions(ctx *sql.Context, hasErr bool) {
+	incVal := int64(1)
+	if hasErr {
+		// for some reason, errors count as two questions
+		incVal = int64(2)
+	}
+	// ignore all errors relating to status updates
```

Comment:
Good note about ignoring errors. That seems appropriate here. It might be helpful to add a similar comment mentioning that there is a race condition here, but we are consciously making the tradeoff to have a small risk of race condition and losing a question count or two instead of taking on the extra expense of a mutex. Alternatively... I saw your comment below about considering spinning up a go routine. If we do want to add locking here, then I think you're right it would make sense to spin up a goroutine at that point. 

I mention that, since I think someone looking at this code in the future might wonder why there is no locking, so it's nice to encode that decision in a comment.

Reply:
Btw... any more context on how you saw the count being incremented twice for errors? I'm not seeing the same behavior. I do see an increment of 2, but I think one is for the error statement and one is for `SHOW STATUS`:

```
mysql> show status like 'Questions';
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Questions     | 19    |
+---------------+-------+
1 row in set (0.00 sec)

mysql> show status like 'Questions';
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Questions     | 20    |
+---------------+-------+
1 row in set (0.00 sec)

mysql> asdf;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'asdf' at line 1
mysql> show status like 'Questions';
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Questions     | 22    |
+---------------+-------+
1 row in set (0.00 sec)
```

Reply:
This is what I see from MySQL:
```sql
mysql> show status like "questions";
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Questions     | 41    |
+---------------+-------+
1 row in set (0.0010 sec)

mysql> asdf;
ERROR: 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'asdf' at line 1

mysql> show status like "questions";
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Questions     | 44    |
+---------------+-------+
1 row in set (0.0012 sec)
```

I think it might be running a query under the hood to log errors? or... it's some windows quirk.
I'll just have it increment by one now, and can just update it whenever I figure out why errors are counted twice on my machine.